netbox_librenms_plugin/utils.py

```
@@ -949,27 +1120,63 @@ def get_librenms_device_id(obj, server_key: str = "default", *, auto_save: bool
+        if int_id is None:
+            return None
+        # Normalise a string-stored id back to the DB so later queries use a plain int.
+        if auto_save and isinstance(value, str):
+            cf_value[server_key] = int_id
+            obj.custom_field_data["librenms_id"] = cf_value
+            obj.save(update_fields=["custom_field_data"])
+        return int_id
     return None
 
 
+def is_legacy_librenms_id(value) -> bool:
```

Flake8 (7.3.0)

[error] redefinition of unused 'is_legacy_librenms_id' from line 1050

(F811)